src/engraving/layout/layoutmeasure.cpp

```diff
@@ -501,7 +515,28 @@ static bool breakMultiMeasureRest(const LayoutContext& ctx, Measure* m)
         }
     }
 
+    // Break for annotations found mid-way into the previous measure
+    if (prevMeas) {
+        for (Segment* s = prevMeas->first(); s; s = s->next()) {
+            for (EngravingItem* e : s->annotations()) {
+                if (!e->visible()) {
+                    continue;
+                }
+                if ((e->isRehearsalMark()
+                     || e->isTempoText()
+                     || ((e->isHarmony() || e->isStaffText() || e->isSystemText() || e->isTripletFeel() || e->isPlayTechAnnotation()
```

Comment:
This block of code is impossible to read/debug. I think you could use static std::set/unordered_set here instead of adding a lot of checks to this condition